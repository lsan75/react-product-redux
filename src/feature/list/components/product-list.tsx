import * as React from 'react'
import { IProduct } from '../../../store/product/product'

import './product-list.css'

export interface IProps {
  productList: IProduct[]
  selectProduct: (product: IProduct) => void
}

export function ProductList(props: IProps) {

  const productItems = props.productList.map(item => {
    const className = `ProductList__Item ${ item.selected ? 'selected' : ''}`
    const click = () => props.selectProduct(item)
    return (
      <li key={item.name} className={className} onClick={click}>
        <strong className="ProductList__Item__Name">{item.name}</strong>
        <span className="ProductList__Item__Price">{item.price}&euro;</span>
      </li>
    )
  })

  return (
    <ul className="ProductList">
      {productItems}
    </ul>
  )
}