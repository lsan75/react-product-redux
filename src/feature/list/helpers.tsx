import { IProduct } from '../../store/product/product'

export const productList: IProduct[] = [
  {
    name: 'aaa',
    price: 10,
    selected: true
  },
  {
    name: 'bbb',
    price: 20,
    selected: false
  },
  {
    name: 'ccc',
    price: 50
  }
]