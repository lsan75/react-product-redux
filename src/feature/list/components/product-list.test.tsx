import * as React from 'react'

import * as Adapter from 'enzyme-adapter-react-16'

import { configure, mount, ReactWrapper } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'

import { IProps, ProductList } from './product-list'

import { productList } from '../helpers'

describe('ProductList', () => {
  configure({ adapter: new Adapter() })
  let reactOutput: ReactWrapper

  const selectProduct = jest.fn()

  const mountComp = () => {
    return mount<IProps>(
      <ProductList
        productList={productList}
        selectProduct={selectProduct}
      />
    )
  }

  afterEach(() => {
    reactOutput.unmount()
  })

  it('should match the snapshot', () => {
    reactOutput = mountComp()
    expect(mountToJson(reactOutput)).toMatchSnapshot()
  })

  it('should select a product', () => {
    reactOutput = mountComp()
    const li = reactOutput.find('ul').childAt(1)

    li.simulate('click')

    expect(selectProduct).toBeCalledWith({
      name: 'bbb',
      price: 20,
      selected: false
    })
  })
})
