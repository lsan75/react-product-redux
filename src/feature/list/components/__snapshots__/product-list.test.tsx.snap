// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList should match the snapshot 1`] = `
<ProductList
  productList={
    Array [
      Object {
        "name": "aaa",
        "price": 10,
        "selected": true,
      },
      Object {
        "name": "bbb",
        "price": 20,
        "selected": false,
      },
      Object {
        "name": "ccc",
        "price": 50,
      },
    ]
  }
  selectProduct={[MockFunction]}
>
  <ul
    className="ProductList"
  >
    <li
      className="ProductList__Item selected"
      key="aaa"
      onClick={[Function]}
    >
      <strong
        className="ProductList__Item__Name"
      >
        aaa
      </strong>
      <span
        className="ProductList__Item__Price"
      >
        10
        €
      </span>
    </li>
    <li
      className="ProductList__Item "
      key="bbb"
      onClick={[Function]}
    >
      <strong
        className="ProductList__Item__Name"
      >
        bbb
      </strong>
      <span
        className="ProductList__Item__Price"
      >
        20
        €
      </span>
    </li>
    <li
      className="ProductList__Item "
      key="ccc"
      onClick={[Function]}
    >
      <strong
        className="ProductList__Item__Name"
      >
        ccc
      </strong>
      <span
        className="ProductList__Item__Price"
      >
        50
        €
      </span>
    </li>
  </ul>
</ProductList>
`;
